File: 647-PalindromicSubstrings/647-PalindromicSubstrings.java
// Last updated: 2/28/2026, 4:27:11 PM
class Solution {
    public int countSubstrings(String s) {

        if(s == null || s.length() == 0) return 0;

        if(s.length() == 1) return 1;

        int count = 0;

        for(int i = 0; i < s.length(); i++){
            count += expandFromCenter(s, i, i);
            count += expandFromCenter(s, i,i+1);
        }

        return count; 
        
    }

    public int expandFromCenter(String s, int left, int right){

        if(s == null || s.length() == 0) return 0;

        int count = 0;
        while(left >= 0 && right < s.length() && s.charAt(left) == s.charAt(right)){
            left--;
            right++;
            count++;
        }

        return count;

    }
}

File: 647-PalindromicSubstrings/647-PalindromicSubstrings.py
# Last updated: 7/12/2026, 6:18:56 PM
class Solution:
    def countSubstrings(self, s: str) -> int:
        n = len(s) 

        if not s or len(s) == 0: return 0

        if len(s) == 1: return 1
        
        ans = 0
        for i in range(n):
            ans += self.expandFromCenter(s,i, i)
            ans += self.expandFromCenter(s,i, i + 1)
        
        return ans




    def expandFromCenter(self, s, i, j):
        n = len(s)
        count = 0

        while i >= 0 and j < n and s[i] == s[j]:
            i-=1
            j+=1
            count += 1
        
        return count
        

        